### PR TITLE
Remove hazelcast-all module

### PR DESCRIPTION
**Pull request by frant-hartm**
The EE counterpart: hazelcast/hazelcast-enterprise#4128

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -921,104 +920,6 @@
...
-                                <id>attach-javadocs</id>
-                                <goals>
-                                    <goal>jar</goal>
-                                </goals>
-                            </execution>
-                        </executions>
-                    </plugin>
-                </plugins>
-            </build>
-        </profile>
```
Comment (kwart) [inline pom.xml:1013]: AFAIK, we still need the `release` and `release-snapshot` profiles for the deployments. CC @olukas Could you confirm? Not sure about the `zip` one.

**Review by kwart — CHANGES_REQUESTED** [review]
Could you prepare also the EE counterpart PR? The `hazelcast-enterprise-all` depends on the `hazelcast-all`.

**Comment by frant-hartm** [comment]
This is the EE counterpart https://github.com/hazelcast/hazelcast-enterprise/pull/4128.

File: pom.xml (hunk 1 above)
Comment (mmedenjak) [reply pom.xml:1013]: Yes, I'd say at least `release`. He did mention it on release night as something he was running - https://hazelcast.slack.com/archives/C026A3BMTQD/p1626204511408900?thread_ts=1626204309.408000&cid=C026A3BMTQD

Comment (mmedenjak) [reply pom.xml:1013]: Regarding `release-snapshot`, I don't see much reference to it on slack (which doesn't mean it's not used). Only this - https://github.com/hazelcast/hazelcast/issues/14989

**Review by mmedenjak — APPROVED** [review]
Looks good from my side, only controversial change was the pom.xml which Josef already pointed out to.

File: pom.xml (hunk 1 above)
Comment (kwart) [reply pom.xml:1013]: The profile is used in deploy jobs (from dev branches). E.g. http://jenkins.hazelcast.com/job/Hazelcast-master-deploy/configure

**Review by kwart — APPROVED** [review]

**Pull request merged**